File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/dto/MappedSuperclassFragmentTest.java
package org.babyfish.jimmer.sql.dto;

import org.babyfish.jimmer.sql.model.inheritance.Role;
import org.babyfish.jimmer.sql.model.inheritance.dto.RoleWithMappedSuperclassFragmentView;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class MappedSuperclassFragmentTest {

    @Test
    public void testMappedSuperclassFragment() {
        LocalDateTime createdTime = LocalDateTime.of(2026, 7, 18, 12, 0);
        LocalDateTime modifiedTime = createdTime.plusHours(1);
        RoleWithMappedSuperclassFragmentView view = new RoleWithMappedSuperclassFragmentView();
        view.setId(1L);
        view.setName("admin");
        view.setCreatedTime(createdTime);
        view.setModifiedTime(modifiedTime);

        Role role = view.toEntity();
        assertEquals(1L, role.getId());
        assertEquals("admin", role.getName());
        assertEquals(createdTime, role.getCreatedTime());
        assertEquals(modifiedTime, role.getModifiedTime());
    }
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/dto/MappedSuperclassFragmentTest.kt
package org.babyfish.jimmer.sql.kt.dto

import org.babyfish.jimmer.sql.kt.model.inheritance.dto.RoleWithMappedSuperclassFragmentView
import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertEquals

class MappedSuperclassFragmentTest {

    @Test
    fun testMappedSuperclassFragment() {
        val createdTime = LocalDateTime.of(2026, 7, 18, 12, 0)
        val modifiedTime = createdTime.plusHours(1)
        val view = RoleWithMappedSuperclassFragmentView(
            id = 1L,
            name = "admin",
            createdTime = createdTime,
            modifiedTime = modifiedTime
        )

        val role = view.toEntity()
        assertEquals(1L, role.id)
        assertEquals("admin", role.name)
        assertEquals(createdTime, role.createdTime)
        assertEquals(modifiedTime, role.modifiedTime)
    }
}
